trigger.js

```diff
@@ -215,39 +224,76 @@ class TriggerHappy {
-                update.x = target.center.x - tw;
-                update.y = target.center.y - th;
-                return true;
+            const tx = target.x;
+            const ty = target.y;
+            const tw = target.w;
+            const th = target.h;
+            const tgw = target.data.width; // target token width in grid units
+            const tgh = target.data.height; // target token height in grid units
+                // test motion vs token diagonals
+            if (tgw > 1 && tgh > 1 && tgw * tgh > 4) {
+                // big token so do boundary lines
+                var intersects = ( motion.intersectSegment([tx,      ty,      tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx,      ty + th])
+                                || motion.intersectSegment([tx,      ty,      tx,      ty + th]))
+            } else  // just check the diagonals
+                var intersects = (motion.intersectSegment([tx,       ty,      tx + tw, ty + th])
+                               || motion.intersectSegment([tx,       ty + th, tx + tw, ty     ]));
+            if (intersects) {
+                if (tgw === 1 && tgh === 1) { // simple case size 1 target, return straight away.
+                    update.x = target.center.x - tkw;
+                    update.y = target.center.y - tkh;
+                    return true;
+                }
+                
+                let corners = [];
+                const gridSize = canvas.grid.size;
+                // Create a grid of the squares covered by the target token
+                [...Array(tgw).keys()].forEach(xc => [...Array(tgh).keys()].forEach(yc => 
+                    corners.push({x: target.data.x + xc * gridSize, y: target.data.y + yc * gridSize})
+                ))
+
+                // Find the closest square to the token start position that intersets the motion
+                const closest = corners.sort((a, b) => 
+                    Math.hypot(token.x + tkw - (a.x + gridSize / 2), token.y  + tkh - (a.y + gridSize / 2)) - Math.hypot(token.x + tkw - (b.x + gridSize / 2), token.y  + tkh - (b.y + gridSize / 2)));
+                for (let corner of closest) {
+                    if (motion.intersectSegment([corner.x, corner.y, corner.x + gridSize, corner.y + gridSize])
+                     || motion.intersectSegment([corner.x, corner.y + gridSize, corner.x + gridSize, corner.y])) {
+                        update.x = corner.x;
+                        update.y = corner.y;;
+                        return true;
+                    }
+                };
+                console.warn("Ttrigger Happy | Help me the universe is non-euclidean");
             }
         }
         return true;
     }
-    _onPreUpdateToken(scene, embedded, update, options, userId) {
+    _onPreUpdateToken(scene, id, update) {

```

Comment:
This was changed to fit the new 0.5.4 arguments : https://github.com/kakaroto/fvtt-module-trigger-happy/commit/dc603d299c5f819050655ce33eafccc22fbbb67c
Since we don't use `embedded` or `id`, I didn't change the minimumCoreVersion

Reply:
This remains to be reverted back.